js/vapid.js

```diff
@@ -90,6 +71,30 @@ var vapid = {
         return reply;
     },
 
+    _array_to_str: function(array) {
+        /* convert a ByteArray into a string
+         */
+        return String.fromCharCode.apply(null, new Uint8Array(array));
+    },
+
+    urlsafe: function(str) {
+        return str
+            .replace(/\+/g, "-")
+            .replace(/\//g, "_");
+    },
+
+    url_btoa: function(data) {
```

Comment:
These `btoa` and `atob` names confuse me. Maybe `binaryToString` and `stringToBinary` ?